zebra-chain/src/types/amount.rs

```diff
@@ -127,6 +132,16 @@ impl From<Amount<NonNegative>> for u64 {
     }
 }
 
+impl<C> From<Amount<C>> for jubjub::Fr {
+    fn from(a: Amount<C>) -> jubjub::Fr {
+        if a.0 < 0 {
```

Comment:
```suggestion
        // TODO: this isn't constant time -- does that matter?
        if a.0 < 0 {
```

Reply:
Consider requesting an `Fr::from_signed` method. zcashd could use that as well I think.